hal/shared/mcp23s17.cpp

```diff
@@ -135,9 +135,19 @@ int Mcp23s17::reset(bool verify) {
     hal_gpio_write(IOE_RST, 1);
     HAL_Delay_Milliseconds(10);
 
+#if HAL_PLATFORM_MCP23S17_MIRROR_INTERRUPTS || HAL_PLATFORM_MCP23S17_OPEN_DRAIN_INT
+    uint8_t iocon {};
+
 #if HAL_PLATFORM_MCP23S17_MIRROR_INTERRUPTS
-    CHECK(writeRegister(IOCON_ADDR[0], 1u << 6));
+    iocon |= 1u << 6;
+#endif
+#if HAL_PLATFORM_MCP23S17_OPEN_DRAIN_INT
+    iocon |= 1u << 2;
```

Comment:
Nit: magic numbers. We have plenty of those here, but if we can define self-describing constants for these would be great.

Reply:
I generally agree, but in a case where the descriptive flag is right next to the magic number that's used once, it's not adding much. 🤷‍♂️